### PR TITLE
Fixes #30175,#30195 - Moving Quota validation to before_save in Ovirt CR

### PR DESCRIPTION
**Pull request by yifatmakias**

### CODE REVIEW

**Comment by shiramax** [comment]
thanks @yifatmakias !
Tests failures are related, also, can you try change the commit message to something more descriptive? maybe "Moving Quota validation to before_save in Ovirt CR"

**Review by shiramax — CHANGES_REQUESTED** [review]
thanks @yifatmakias !
Tests failures are related, also, can you try change the commit message to something more descriptive? maybe "Moving Quota validation to before_save in Ovirt CR"

File: test/controllers/hosts_controller_test.rb [hunk 1]
```diff
@@ -1345,7 +1345,7 @@ class Host::Valid < Host::Managed; end
 
   test "#host update shouldn't diassociate from VM" do
     hostgroup = FactoryBot.create(:hostgroup, :with_environment, :with_subnet, :with_domain, :with_os)
-    compute_resource = compute_resources(:ovirt)
+    compute_resource = compute_resources(:one)
     compute_resource.update(:locations => hostgroup.locations, :organizations => hostgroup.organizations)
```
Comment (shiramax) [inline test/controllers/hosts_controller_test.rb:1353]: why did you change :ovirt to :one?

Comment (yifatmakias) [reply test/controllers/hosts_controller_test.rb:1353]: Because I understood that the purpose of the test is to check that the host update does not disassociate from VM and therefore it does not matter which compute resource to check. also, I noticed that all the other tests here are working with this compute resource compute_resources(:one) and not with ovirt.
If I misunderstood please let me know and I will add the quota and client part so the test works as it is.

Comment (shiramax) [reply test/controllers/hosts_controller_test.rb:1353]: @yifatmakias  It shouldn't matter, But I'm wondering what error did you get the cause change the CR? and how these two compute resources are different?

Comment (yifatmakias) [reply test/controllers/hosts_controller_test.rb:1353]: I received this error:
Error:
HostsControllerTest#test_0083_#host update shouldn't diassociate from VM:
OvirtSDK4::ConnectionError: Can't send request: Couldn't resolve host name
    app/models/compute_resources/foreman/model/ovirt.rb:522:in `client'
    app/models/compute_resources/foreman/model/ovirt.rb:494:in `validate_quota'
    test/controllers/hosts_controller_test.rb:1349:in `block in <class:HostsControllerTest>'
The two compute resources are different because compute_resources(:ovirt) is an ovirt cr and compute_resources(:one) is a libvirt cr.

Comment (shiramax) [reply test/controllers/hosts_controller_test.rb:1353]: @yifatmakias, 
It seems like it was on ovirt on purpose.

Comment (yifatmakias) [reply test/controllers/hosts_controller_test.rb:1353]: @shiramax Ok so I will change it back to ovirt and deal with the error.

File: test/controllers/hosts_controller_test.rb [hunk 2]
```diff
@@ -1345,7 +1345,7 @@ class Host::Valid < Host::Managed; end
 
   test "#host update shouldn't diassociate from VM" do
     hostgroup = FactoryBot.create(:hostgroup, :with_environment, :with_subnet, :with_domain, :with_os)
-    compute_resource = compute_resources(:ovirt)
+    compute_resource = compute_resources(:one)
```
Comment (shiramax) [inline test/controllers/hosts_controller_test.rb]:
```suggestion
    require 'fog/ovirt/models/compute/quota'
    compute_resource = compute_resources(:ovirt)
    quota = Fog::Ovirt::Compute::Quota.new({ :id => '1', :name => "Default" })
    client_mock = mock.tap { |m| m.stubs(datacenters: [], quotas: [quota]) }
    compute_resource.stubs(:client).returns(client_mock)
```

**Review by shiramax — CHANGES_REQUESTED** [review]

**Review by shiramax — APPROVED** [review]

**Review by ares — APPROVED** [review]
Thanks @yifatmakias and @shiramax, +1 for the test changes. merging!
